### DISCUSSION TITLE
dynamic api routes return same data

### DISCUSSION THREAD

**Discussion by qkreltms** [vercel/next.js#36470] [Help] [answered]
I made 'classroom/courses/:courseId/', 'classroom/courses/:courseId/chapters' routers using api folder structure

<img width="408" alt="image" src="https://user-images.githubusercontent.com/25196026/165252443-78f5e509-6c98-47e8-adc6-5f23c90aa008.png">

When I do this
```
      const { data } = await http.get('api/classroom/courses/210052');  // http.get is fetch api wrapper
      console.log(data); // expected: { data: {title: 'abc', ...}}, result: { data: [...]}

      const { data2 } = await http.get('api/classroom/courses/210052/chapters');
      console.log(data2); // expected: { data: [...]}, result: { data: [...]}
```

I expected that these are must return different data, but these are return same data.

Q)
1. Am I doing right way to make 'classroom/courses/:courseId/', 'classroom/courses/:courseId/chapters' routers using folder structure?
2. How to make these routers return different data as I expected?

**Comment by icyJoseph** [comment]
Hi,

Could you try like this:

```shell
pages/api/
└── posts
    ├── [...postid].ts
    └── [postid]
        └── comments.ts
```

Because then:

```shell
###
GET http://localhost:3000/api/posts/2001
```

This returns:

```json
{
  "query": {
    "postid": [
      "2001"
    ]
  }
}
```

While this one: 

```shell
###
GET http://localhost:3000/api/posts/2001/comments
```

Returns:

```json
{
  "query": {
    "postid": "2001"
  }
}
```

I guess for `pages/api/posts/[...postid].ts` you'd need to make sure `postid` from `query` is a length 1 list, and validate the single element against your database, while in `pages/api/posts/[postid]/comments.ts` you just validate `postid` from `query`.

**Reply by qkreltms** [reply]
Should I do like this?
I did change a file name [courseId].ts => [...courseId].ts
<img width="407" alt="image" src="https://user-images.githubusercontent.com/25196026/165261396-dfb30058-44ec-41b8-b5cb-f06693db3b39.png">

It returns expected data at first time, but when I refresh page it returns wrong data.


1)
Request URL: http://localhost:8080/api/classroom/courses/210052
Request Method: GET
Status Code: 200 OK
Remote Address: [::1]:8080
Referrer Policy: strict-origin-when-cross-origin

2)
Request URL: http://localhost:8080/api/classroom/courses/210052/chapters
Request Method: GET
Status Code: 200 OK
Remote Address: [::1]:8080
Referrer Policy: strict-origin-when-cross-origin

**Reply by icyJoseph** [reply]
Could you explain a bit more of:

> It returns expected data at first time, but when I refresh page it returns wrong data.

How could the API call return wrong data after a refresh. Scary.

**Reply by qkreltms** [reply]
1. Restart react-next by typing `npm run start`
2. refresh browser, each routers return different data with status code 200 and 304.
3. refresh browser, each routers return same data with status code 200 and 304.
4. refresh browser (command + Shift + R), each routers return same data with status code 200.

**Comment by qkreltms** [comment, marked as answer]
I found out it's due to next-connect.

I did 
// _handler.ts
```js
const handler = nc({
  attachParams: true,
});

handler.use(cors());

export default handler;
```
// [courseId].ts, chapter.ts
```js
import handler from './_handler'
hander.get(...)
```
------
I was able to resolve it by doing like this
// [courseId].ts, chapter.ts
```js
const handler = nc();
hander.get(...)
```

**Reply by icyJoseph** [reply]
Aha, kind of an important detail.
